### ISSUE TITLE
Controlling selectingDate from the outside

### ISSUE THREAD

**Issue by mfolnovic** [Hacker0x01/react-datepicker#1614]
I'm trying to build a date range component that works a bit differently than your example. 

This is what it looks like at the moment:
![image](https://user-images.githubusercontent.com/20919/51613784-83153680-1f24-11e9-8eb9-54171b2bd6c8.png)

As you can see, the start date is 01/09/2019 and the end date is 02/13/2019. Because those two are separate datepickers, they don't know about each others `selectingDate`.

I was wondering if there's a way to have:
- a callback when `selectingDate` changes
- a prop `selectingDate` that would serve as fallback in case `selectingDate` in state isn't set.

Basically, I'm trying to control what's styled as in range while hovering over days.

If that's okay, I could provide PR for this.

**Comment by mfolnovic** [comment]
@martijnrusschen what do you think? :)

**Comment by mfolnovic** [comment]
@martijnrusschen ping :)

**Comment by mfolnovic** [comment]
ping :)

**Comment by goszczynskip** [comment]
Another usecase that I see where your propsal could be usefull is offset range selecting. With controlled approach we can "highlight" ranges that aren't directly hovered dates during selection phase eg. one week in advance. 

Without controlled state another approach would be to do it like react-dates. They are allowing offset property which calculates proper date while user is moving mouse over component.

@mfolnovic are you still interested in creating PR for controlled state?

**Comment by mfolnovic** [comment]
@goszczynskip I've switched to https://github.com/tresko/react-datepicker/tree/master/packages/hooks, sorry :)
